lib/ipaddress/ipv6.rb

```diff
@@ -315,8 +315,111 @@ def reverse
       to_hex.reverse.gsub(/./){|c| c+"."} + "ip6.arpa"
     end
     alias_method :arpa, :reverse
+    
+    #
+    # Returns a new IPv6 object which is the result 
+    # of advancing this IP address by a given value.
+    # In other words, this arithmetically adds IP addresses.
+    #
+    # Will raise an error if the resulting address is in a different subnet,
+    # except validating is set to false.
+    #
+    # Example:
+    #
+    #   ip = IPAddress::IPv6.new("fc42:1337::/64")
+    #   ip.add(5).to_string
+    #     #=> "fc42:1337::5/64"
+    def add(oth, validating=true)
+      oth = oth.to_i if oth.kind_of? IPAddress::IPv6 # oth shall be integer
+      
+      new_obj = self.class.parse_u128(self.to_i + oth, prefix)
+      
+      if validating and self.network_u128 != new_obj.network_u128
+        raise ArgumentError, "Subnet (/#{@prefix}) is not large enough."
```

Comment:
what do you think about changing this to a ```RuntimeError```.  The reason why I suggest this is that there isn't a problem with argument per se, rather you have reached a state of exhaustion.